nearcache/config.go

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nearcache
+
+import (
+	"math"
+
+	"github.com/hazelcast/hazelcast-go-client/internal/check"
+	ihzerrors "github.com/hazelcast/hazelcast-go-client/internal/hzerrors"
+)
+
+const (
+	defaultMaxEntryCount            = 10_000
+	defaultEvictionPolicy           = EvictionPolicyLRU
+	defaultStoreInitialDelaySeconds = 600
+	defaultStoreIntervalSeconds     = 600
+	defaultMemoryFormat             = InMemoryFormatBinary
+)
+
+// Config is the Near Cache configuration.
+type Config struct {
+	invalidateOnChange *bool
+	// Name is the name of this Near Cache configuration.
```

Comment:
It's actually not the name of the near cache, because the name can have wildcards, like `mymap*` which matches near caches for `mymap`, `mymap1`, etc.